common/scripted_effects/iisc_corps_effects.txt

```diff
@@ -47,6 +42,178 @@ iisc_get_overlord = {
 	}
 }
 
+# this = corp
+# hq set and exists
+iisc_handle_corp_connection_to_overlord = {
+	if = {
+		limit = {
+			OR = {
+				NOT = {
+					exists = event_target:iisc_hq_planet.owner
+				}
+				event_target:iisc_hq_planet.owner = {
+					iisc_can_have_space_corp = no
+				}
+				event_target:iisc_hq_planet.owner = {
+					is_subject = yes
+				}
+			}
+		}
+		iisc_destroy_corp = yes
+	}
+	else_if = {
+		# the owner exists and can have corps
+		limit = {
+			# same as a NAND without the NOTs but this way it's easier to understand
+			OR = {
+				NOT = {
+					exists = overlord
+				}
+				NOT = {
+					overlord = {
+						is_same_value = event_target:iisc_hq_planet.owner
+					}
+				}
+			}
+		}
+		# the corp is not connected
+		# we need to make it subject to the owner of the planet
+		# the owner of the planet should not be already a subject thanks to the on_becoming_subject
+		set_subject_of = {
+			who = event_target:iisc_hq_planet.owner
+			subject_type = iisc_corp_vassal
+		}
+		# if necessary in the future: scope to overlord and
+		# iisc_count_corps = yes
+		# TODO add an event with a window?
+	}
+}
+
+# this = corp
+# overlord = the new overlord
+iisc_disband_corp = {
+	if = {
+		limit = {
+			is_country_type = iisc_space_corporation
+		}
+		# just a redundant check
+		if = {
+			limit = {
+				exists = overlord
+			}
+			# load (old) hq
+			# remove hq deposit
+			overlord = {
+				random_subject = {
+					limit = {
+						iisc_can_have_space_corp = yes
+						any_owned_planet = {
+							has_planet_flag = iisc_hq_of_@prevprevprev
+						}
+					}
+					random_owned_planet = {
+						limit = {
+							has_planet_flag = iisc_hq_of_@prevprevprev
+						}
+						save_event_target_as = iisc_hq_planet
+						remove_planet_flag = iisc_hq_of_@prevprevprev
+						prevprevprev = {
+							iisc_remove_hq_deposit_from_corp = yes
+						}
+					}
+				}
+			}
```

Comment:
Some of these checks seem ultimately unnecessary, that is, scoping to the overlord, then a random subject (the former owner of the corporation), then a random planet of theirs, to get to the HQ? This actually seems needlessly restrictive, since these checks mean we can _only_ use this effect for removing a corporation when their owner has becoming a vassal and the corp is now invalid. But what if in the future we decide we would like to remove a corp and have their money distributed in some other scenario (where their owner hasn't been vassalized)?

In short it seems to me that this effect could just be done with a `random_planet` with the `iisc_hq_of_@` flag. This is slower but this event isn't running regularly, so I'm not sure it would be a serious issue.